mathesar_ui/src/components/select/Select.svelte

```diff
@@ -24,25 +24,71 @@
   export let ariaLabel: string = null;
 
   let isOpen = false;
+  let currentIndex = 0;
+  let erasedIndex = 0;
+  let t = 0;
 
   function setValue(opt: SelectOption) {
     value = opt;
     dispatch('change', {
       value,
     });
     isOpen = false;
+    erasedIndex = 0;
+    currentIndex = 0;
   }
-
+  
   function setOptions(opts: SelectOption[]) {
     if (!value && opts.length > 0) {
       setValue(opts[0]);
     }
   }
+  
+  function hoveredItem(index){
+      let items= [...document.getElementById('select-value-'+selectId).getElementsByTagName('li')];
+      if(currentIndex == items.length - 1 && index > 0){
+        currentIndex = 0;
+        index = 0;
+      }else if(currentIndex == 0 && index < 0){
+        currentIndex = items.length;
+      }else if(currentIndex == 0 && index > 0){
+        if(t <1){
+          currentIndex = -1;
+        }else if (t > 1){
+          currentIndex = 0;
+        }
+        t += 1; 
+      }
+      if(erasedIndex >= 1){
+        let hoveredElement= document.querySelector(".hovered");
+        hoveredElement.classList.remove("hovered");
+      }
+      currentIndex = currentIndex + index; 
+      items[currentIndex].classList.add("hovered");
+      erasedIndex = 1; 
+    }
 
-  $: setOptions(options);
+  function keyAccessibility(e){
+      switch (e.key){
+        case "ArrowDown":
+          hoveredItem(1);
+          break;
+        case "ArrowUp":
+          hoveredItem(-1);
+          break;
+        case "Escape":
+          isOpen = false;
+          break;
+        case "Enter":
+          if (options.length == 0) break;
+          setValue(options[currentIndex]);
+          break;
```

Comment:
* The trigger for dropdown's default behaviour for enter is to switch between open/close. i.e when enter key is pressed on the button, it toggles the state.
* Here, isOpen=false is explicitly mentioned within setValue.

When enter is clicked two things happen:
* Firstly, isOpen is set to false from within setValue.
* Then the button's default behaviour is triggered. It sees that the value for isOpen is false and then switches it to isOpen=true. So the dropdown actually closes and reopens.

It is best to not change isOpen=false when the action happens through keyDown. It should still do it for click events.

Reply:
Thanks for the detailed description, it was very helpful for understanding how it works. I was able to figure how to close the dropdown with 'Enter', however, I still used isOpen = false. I also used .preventDefault to manage the scrollbehaviour.  It prevents the page from scrolling while using the arrow keys. Here is the code: 
```
case "Enter":
          e.preventDefault();
          if (options.length == 0) break;
          value = options[currentIndex];
          dispatch('change', {
            value,
          });
          isOpen = false;
          break;
```
I can try and research other ways to prevent the use of 'isOpen = false'. But will take more time. Let me know.

Reply:
@mr-gabe49 I think it should be fine, if you got it to work as expected.

Regarding scroll behaviour, there are two things to consider:
1. The whole page scrolls when keydown is pressed. This can be fixed by preventDefault, as you've done.
2. When there is a scrollbar within the dropdown (if the number of elements are more), upon keydown an element which is not in view could become active. It should be made visible to the user.

Handling the 2nd point can be done separately. I would like to get this PR merged early. We can handle each specific requirement incrementally.

Reply:
@pavish I've handled the changes requested. Only thing left would be managing the scrollbar within dropdown, which we decided to do separately. I would like to be assigned that feature. Would raising an issue be necessary?